include/tests_authentication

```diff
@@ -352,50 +352,50 @@
-        if [ -z "${FIND}" -o "${FIND}" = "0" ]; then
-            LogText "Result: number of minimum rounds used by the encryption algorithm is not configured"
-            Display --indent 2 --text "- Checking minimum group password hashing rounds" --result "${STATUS_DISABLED}" --color YELLOW
-            ReportSuggestion "${TEST_NO}" "Configure minimum encryption algorithm rounds in /etc/login.defs"
-            AddHP 0 2
-        elif [ "${FIND}" -lt 5000 ]; then
-            LogText "Result: low number of minimum encryption algorithm rounds found: ${FIND}"
-            PASSWORD_MINIMUM_ROUNDS=${FIND}
-            Display --indent 2 --text "- Group password hashing rounds (minimum)"  --result "${STATUS_SUGGESTION}" --color YELLOW
-            AddHP 1 2
+        SHA_CRYPT_MIN_ROUNDS_FIND=$(${GREPBINARY} "^SHA_CRYPT_MIN_ROUNDS" ${ROOTDIR}etc/login.defs | ${AWKBINARY} '{ if ($1=="SHA_CRYPT_MIN_ROUNDS") { print $2 } }')
+        SHA_CRYPT_MAX_ROUNDS_FIND=$(${GREPBINARY} "^SHA_CRYPT_MAX_ROUNDS" ${ROOTDIR}etc/login.defs | ${AWKBINARY} '{ if ($1=="SHA_CRYPT_MAX_ROUNDS") { print $2 } }')
+        SHA_CRYPT_ROUNDS=0
+
+        if [ -n "${SHA_CRYPT_MIN_ROUNDS_FIND}" -a -n "${SHA_CRYPT_MAX_ROUNDS_FIND}" ]; then
+            if [ ${SHA_CRYPT_MIN_ROUNDS_FIND} -lt ${SHA_CRYPT_MAX_ROUNDS_FIND} ]; then
+                SHA_CRYPT_ROUNDS=${SHA_CRYPT_MIN_ROUNDS_FIND}
+            else
+              SHA_CRYPT_ROUNDS=${SHA_CRYPT_MAX_ROUNDS_FIND}
+            fi
+        elif [ -z "${SHA_CRYPT_MIN_ROUNDS_FIND}" -a -n "${SHA_CRYPT_MAX_ROUNDS_FIND}" ]; then
+            SHA_CRYPT_ROUNDS=${SHA_CRYPT_MAX_ROUNDS_FIND}
+        elif [ -n "${SHA_CRYPT_MIN_ROUNDS_FIND}" -a -z "${SHA_CRYPT_MAX_ROUNDS_FIND}" ]; then
+            SHA_CRYPT_ROUNDS=${SHA_CRYPT_MIN_ROUNDS_FIND}
         else
-            LogText "Result: number of encryption algorithm rounds is ${FIND}"
-            PASSWORD_MINIMUM_ROUNDS=${FIND}
-            Display --indent 2 --text "- Group password hashing rounds (minimum)" --result CONFIGURED --color GREEN
-            AddHP 2 2
+            SHA_CRYPT_ROUNDS=0
         fi
 
-        LogText "Test: Checking SHA_CRYPT_MAX_ROUNDS option in ${ROOTDIR}etc/login.defs"
-        FIND=$(${GREPBINARY} "^SHA_CRYPT_MAX_ROUNDS" ${ROOTDIR}etc/login.defs | ${AWKBINARY} '{ if ($1=="SHA_CRYPT_MAX_ROUNDS") { print $2 } }')
-        if [ -z "${FIND}" -o "${FIND}" = "0" ]; then
-            LogText "Result: number of maximum rounds used by the encryption algorithm is not configured"
-            Display --indent 2 --text "- Checking maximum group password hashing rounds" --result "${STATUS_DISABLED}" --color YELLOW
-            ReportSuggestion "${TEST_NO}" "Configure maximum encryption algorithm rounds in /etc/login.defs"
+        LogText "Test: Checking SHA_CRYPT_{MIN,MAX}_ROUNDS option in ${ROOTDIR}etc/login.defs"
+        if [ ${SHA_CRYPT_ROUNDS} -eq 0 ]; then
```

Comment:
This comparison breaks if `SHA_CRYPT_ROUNDS` is `potato`. The original one also, but just wondering if there should be safeguards to check whether it's an integer at all.

Reply:
Maybe use [DigitsOnly()](https://github.com/CISOfy/lynis/blob/701be9852bae4f64adf7b94f874956ffe1fffb54/include/functions#L403-L417) for `SHA_CRYPT_ROUNDS`?

Reply:
Seems reasonably, but `DigitsOnly()` is only used in `plugins/plugin_pam_phase1` so that should probably be another PR where all values are validated.